polyglot/piranha/src/models/rule.rs

```diff
@@ -28,13 +28,13 @@ use super::{
 
 static FEATURE_FLAG_API_GROUP: &str = "Feature-flag API cleanup";
 
-#[derive(Deserialize, Debug, Clone, Hash, PartialEq, Eq, Default)]
```

Comment:
We are removing the option to compare rules for equality here, correct? Is this related to the new type of rules or just an unrelated refactoring/removal? Was this something that worked before but we don't need or something that was broken?

(Should be fine either way, just want to make sure I understand a bit what's happening)

Edit: Ok, looks to me like below we are using a custom definition of equality for `Rule` at least. Wonder if it makes sense to implement `Rule::Eq(...)` somehow.

Reply:
Yes. I think the most correct way would be to defile my own Eq and Hash. I ll do that as a followup .

Reply:
I had defined my own eq, then I thought IT was too much ceremony for such a simple check.  - https://github.com/uber/piranha/pull/208/commits/1100973c2ecae5a07a5b67d85f499aa738820a4f